nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -1220,48 +1221,26 @@ impl DataStore {
         };
 
         Ok(Some(
-            self.inventory_collection_read_all_or_nothing(
+            self.inventory_collection_read_batched(
                 opctx,
                 collection_id,
-                limit,
+                batch_size,
             )
             .await?,
         ))
     }
 
-    /// Attempt to read the given collection while limiting queries to `limit`
-    /// records and returning nothing if `limit` is not large enough.
-    pub async fn inventory_collection_read_all_or_nothing(
+    /// Attempt to read the current collection.
+    ///
+    /// Queries are limited to `batch_size` records at a time, performing
+    /// multiple queries if more than `batch_size` records exist.
+    pub async fn inventory_collection_read_batched(
         &self,
         opctx: &OpContext,
         id: Uuid,
-        limit: NonZeroU32,
+        batch_size: NonZeroU32,
```

Comment:
Same -- if this is `pub`, I'd drop the `batch_size` or make it an optional override.